src/Features/VisualBasic/Portable/ConvertIfToSwitch/VisualBasicConvertIfToSwitchCodeRefactoringProvider.vb

```diff
@@ -1,197 +1,154 @@
-                            Return Nothing
-                        End If
-
-                        Dim leftComparison = DirectCast(left, BinaryExpressionSyntax)
-                        Dim rightComparison = DirectCast(right, BinaryExpressionSyntax)
-                        Dim leftConstant As ExpressionSyntax = Nothing
-                        Dim rightConstant As ExpressionSyntax = Nothing
-                        Dim leftExpression As ExpressionSyntax = Nothing
-                        Dim rightExpression As ExpressionSyntax = Nothing
-
-                        If Not TryDetermineConstant(leftComparison.Right, leftComparison.Left,
-                                                    leftConstant, leftExpression) Then
-                            Return Nothing
-                        End If
-
-                        If Not TryDetermineConstant(rightComparison.Right, rightComparison.Left,
-                                                    rightConstant, rightExpression) Then
-                            Return Nothing
-                        End If
-
-                        If Not _syntaxFacts.AreEquivalent(leftExpression, rightExpression) Then
-                            Return Nothing
-                        End If
-
-                        Dim leftIsLowerBound = IsLowerBound(leftExpression, leftComparison)
-                        Dim rightIsLowerBound = IsLowerBound(rightExpression, rightComparison)
-                        If leftIsLowerBound = rightIsLowerBound Then
-                            Return Nothing
-                        End If
-
-                        If Not SetInitialOrIsEquivalentToSwitchExpression(leftExpression) Then
-                            Return Nothing
-                        End If
-
-                        Dim rangeBounds = If(leftIsLowerBound, (rightConstant, leftConstant), (leftConstant, rightConstant))
-                        Return New Pattern.Range(rangeBounds)
+            Public Overrides Function AsSwitchSectionStatements(operation As IOperation) As IEnumerable(Of SyntaxNode)
+                Return GetStatements(operation.Syntax)
+            End Function
 
+            Private Shared Function GetStatements(node As SyntaxNode) As SyntaxList(Of StatementSyntax)
+                Select Case node.Kind
+                    Case SyntaxKind.MultiLineIfBlock
+                        Return DirectCast(node, MultiLineIfBlockSyntax).Statements
+                    Case SyntaxKind.SingleLineIfStatement
+                        Return DirectCast(node, SingleLineIfStatementSyntax).Statements
+                    Case SyntaxKind.SingleLineElseClause
+                        Return DirectCast(node, SingleLineElseClauseSyntax).Statements
+                    Case SyntaxKind.ElseIfBlock
+                        Return DirectCast(node, ElseIfBlockSyntax).Statements
+                    Case SyntaxKind.ElseBlock
+                        Return DirectCast(node, ElseBlockSyntax).Statements
```

Comment:
there might be a helper you can use from VBSyntaxGenerator for at least a large subset of tehse cases.